backend/src/main/java/com/jobwait/persistence/PostgresController.java

```diff
@@ -0,0 +1,179 @@
+            handleAnswerInsertion("answer_workcontract", setResponse, response);
+        }
+
+        public void jobApplicationCount(Integer response) throws SQLException {
+            ResponseSetter setResponse = s -> s.setInt(2, response);
+            handleAnswerInsertion("answer_jobapplicationcount", setResponse, response);
+        }
+
+        public void jobTitle(String response) throws SQLException {
+            ResponseSetter setResponse = s -> s.setString(2, response);
+            handleAnswerInsertion("answer_jobtitle", setResponse, response);
+        }
+
+        public void yearsOfProExperience(Integer response) throws SQLException {
+            ResponseSetter setResponse = s -> s.setInt(2, response);
+            handleAnswerInsertion("answer_yearsofproexperience", setResponse, response);
+        }
+
+        public void educationLevel(ValidEducationLevel response) throws SQLException {
+            ResponseSetter setResponse = s -> s.setString(2, response.name());
+            handleAnswerInsertion("answer_educationlevel", setResponse, response);
+        }
+    }
+
+    @Override
+    public User updateUserAnswers(User user, Answers answers) {
+        try {
+            System.out.format("answers given: %s", answers.toString());
+            Connection connection = getConnection();
+            connection.setAutoCommit(false);
+
+            InsertAnswerTables insertAnswerTables = new InsertAnswerTables(connection, user.id());
+            insertAnswerTables.jobAcceptDate(answers.jobAcceptDate());
+            insertAnswerTables.jobSearchStartDate(answers.jobSearchStartDate());
+            insertAnswerTables.workModel(answers.workModel());
+            insertAnswerTables.workContract(answers.workContract());
+            insertAnswerTables.jobApplicationCount(answers.jobApplicationCount());
+            insertAnswerTables.jobTitle(answers.jobTitle());
+            insertAnswerTables.yearsOfProExperience(answers.yearsOfProExperience());
+            insertAnswerTables.educationLevel(answers.educationLevel());
+
+            connection.commit();
+            return user;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public User createUserFromAuthId(String authId) {
+        try {
+            Connection connection = getConnection();
+            PreparedStatement statement = connection
+                    .prepareStatement("INSERT INTO users(authhash) VALUES (?) RETURNING *");
+            statement.setString(1, authId);
+            ResultSet resultSet = statement.executeQuery();
+            List<User> users = Util.resultSetRowsToAdaptedRows(resultSet, new PostgresUserAdapter());
+            return Util.assertSingleElement(users);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void deleteUserAndPurgeAnswers(User user) {
```

Comment:
when would we use this? probably ethical to allow users authority over their data, but the entire idea hinges on gathering as much of it as possible...

Reply:
I added this because of the following situation the paranoid lizard living in my brain cooked up:

1. user wants to delete their data because they suddenly hate our website
2. user can't delete data
3. user decides to change all their answers to garbage just to mess with us
4. we now have garbage data
5. sad

I think giving them the option to cleanly delete their data would prevent them from getting pissed off enough to corrupt their existing data.

P.S. if this website ever gets big enough, I think it's illegal to not give users the option to remove their data? idk, probably just some euro law

Reply:
so they would get mad and give us more data? i kind of suspect we are going to get garbage data regardless but we will have to delve into that later. either way i guess you are right we should keep it as an option nevertheless.

perhaps we should be upfront with the fact that all data is anonymous so please dont delete it please please please please

Reply:
I guess I'm concerned about malicious users who get frustrated at us not having the right options available for them but malicious users will be malicious regardless so eh. At the very least a "delete all data associated with this user" might make it easier for us to deal with troll accounts cause we can just wipe anything associated with them easily. Not a super important feature, but not hard to code, so I say we keep it.